client/src/components/auth/Register.js

```diff
@@ -1,83 +1,93 @@
 import React, { Component } from 'react'
 import PropTypes from 'prop-types';
-import './Register.css'
 import classnames from 'classnames';
 import { connect } from 'react-redux';
 import { registeruser } from '../../actions/authActions';
 
 class Register extends Component {
   constructor() {
     super();
-   this.state = {
-     name: '',
-     email: '',
-     password: '',
-     password2: '',
-     errors: {}
-   } 
-   this.onChange = this.onChange.bind(this);
-   this.onSubmit = this.onSubmit.bind(this);
+    this.state = {
+      name: '',
+      email: '',
+      password: '',
+      password2: '',
+      errors: {}
+    }
+    this.onChange = this.onChange.bind(this);
+    this.onSubmit = this.onSubmit.bind(this);
   }
-onChange(e){
-  this.setState({[e.target.name]: e.target.value})
-}
-onSubmit(e){
-  e.preventDefault();
 
-  const newUser = {
-    name: this.state.name,
-    email: this.state.email,
-    password: this.state.password,
-    password2: this.state.password2
-  };
+  onChange(e){
+    this.setState({[e.target.name]: e.target.value});
+  }
 
-  this.props.registeruser(newUser);
+  onSubmit(e){
+    e.preventDefault();
 
-}
+    const newUser = {
+      name: this.state.name,
+      email: this.state.email,
+      password: this.state.password,
+      password2: this.state.password2
+    };
 
-  render() {
-    const {errors} = this.props;
+    this.props.registeruser(newUser, this.props.history);
 
-    const { user } = this.props.auth;
```

Comment:
This line of code is missing compared to Kal's DevConnector: render() {
    const { errors } = this.state;